### ISSUE TITLE
Error in mergePairs when dada objects have zero mergable pairs in a sample

### ISSUE THREAD

**Issue by derele** [benjjneb/dada2#484]
When a dada object contains only invalid denoised pair-combinations for any one sample `mergePairs` fails with (at least to me)  hard to trace errors. Different errors depending on whehter `justConcatenate` is set `TRUE` or `FALSE`. I attached a set of dada and derep objects to demonstrate this. [problem_data.Rda.gz] (https://github.com/benjjneb/dada2/files/2076040/problem_data.Rda.gz)

I fixed it for myself by insterting the following into `mergePairs`, where non-consistent denoised pairs are discarded.   

```r
    ups <- ups[keep, ] ## was an unused varibale originally
    if (nrow(ups)==0) {
        outnames <- c("sequence", "abundance", "forward", "reverse",
                      "nmatch", "nmismatch", "nindel", "prefer", "accept")
        ups <- data.frame(matrix(ncol = length(outnames), nrow = 0))
        names(ups) <- outnames
        return(ups)
    }
```

I stumbled over this issue mostly as a weird edge case in testing of my own package I build ontop of dada2 (I am trying to write a package for working with many different amplicons [from
 different primer pairs] at once). It probably doesn't appear when all samples produce proper data for the amplicon being processed. 

And, I am just realizing that this might be (partially?) a duplicate of https://github.com/benjjneb/dada2/issues/469

**Comment by derele** [comment]
I am realizing that it is a result of the introduction of OMEGA_C, right? Only because of that denoised reads can be discarded. Then the above is a result of no proper forward and reverse combinatoin being left...

**Comment by benjjneb** [comment]
> I am realizing that it is a result of the introduction of OMEGA_C, right? Only because of that denoised reads can be discarded.

Yep, I could see that causing this issue in very small or very low quality samples (i.e. almost no repeated reads).

We'd be happy to take a pull request on this. `mergePairs` should handle this case gracefully.

**Comment by derele** [comment]
Done. Simply returning an empty data.frame in this case. I guess this shoudl be compatible with downstream processing. 
Not sure whether a message or warning should be given. I have not included one for now.

**Comment by derele** [comment]
Now removed a bug and added a message for `verbose=TRUE` see https://github.com/benjjneb/dada2/pull/485

**Comment by benjjneb** [comment]
Should be fixed in 7d8d2df8ad06b47c49fb627d8b8c13a90aa00632

Thanks for the pull request!
